src/reqs.c

```diff
@@ -1376,7 +1376,7 @@ connect_to_upstream (struct conn_s *connptr, struct request_s *request)
                 log_message (LOG_WARNING,
                              "No upstream proxy defined for %s.",
                              request->host);
-                indicate_http_error (connptr, 404,
+                indicate_http_error (connptr, 502,
```

Comment:
is 502 the right error code for this ? according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses one of these 3 could be applicable:

```
502 Bad Gateway

    This error response means that the server, while working as a gateway to get a response needed to handle the request, got an invalid response.

503 Service Unavailable

    The server is not ready to handle the request. Common causes are a server that is down for maintenance or that is overloaded. Note that together with this response, a user-friendly page explaining the problem should be sent. This response should be used for temporary conditions and the Retry-After HTTP header should, if possible, contain the estimated time before the recovery of the service. The webmaster must also take care about the caching-related headers that are sent along with this response, as these temporary condition responses should usually not be cached.

504 Gateway Timeout

    This error response is given when the server is acting as a gateway and cannot get a response in time.
```

Reply:
Yeah, I think that's kinda philosophical question. haproxy for example uses (cf. https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#chapter-1.3.1) 500 for the first case (since it is a configuration error and not temporary until the config is fixed) and 503 for the second. I think 502 should be fine but I can change it if your like.